main.cpp

```diff
@@ -134,238 +153,315 @@ struct CarWash
+
     void storeData(std::string filePath); 
-    //    - retrieve data
     void retrieveData(std::string filePath); 
-    //    - go into power-save mode
     void goIntoPowerSaveMode(); 
 };
 
+void HardDrive::storeData(std::string filePath)
+{
+    filePath = "c:/myfiles/garage.jpg";
+}
+
+void HardDrive::retrieveData(std::string filePath)
+{
+    filePath = "c:/myfiles/resume.doc";
+}
+
+void HardDrive::goIntoPowerSaveMode()
+{
+    
+}
+
 struct Keyboard
 {
-    //5 properties:
-    //    - number of keys (int)
     int numberOfKeys = 100;
-    //    - number of function keys (int)
     int numberOfFunctionKeys = 12;
-    //    - case color (std::string)
     std::string caseColor { "black" };
-    //    - button color (std::string)
     std::string buttonColor { "gray" };
-    //    - width (float)
     float width = 16.0f;
-    //3 things it can do:
-    //    - output button presses
+
     void outputButtonPress(int buttonCode); 
-    //    - display num lock
     void displayNumLock(); 
-    //    - display caps lock
     void displayCapsLock(); 
 };
 
+void Keyboard::outputButtonPress(int buttonCode)
+{
+    buttonCode++;
+}
+
+void Keyboard::displayNumLock()
+{
+
+}
+
+void Keyboard::displayCapsLock()
+{
+
+}
+
 struct Cpu
 {
-    //5 properties:
-    //    - speed (int)
     int speed = 1666;
-    //    - width (float)
     float width = 1.0f;
-    //    - height (float)
     float height = 1.0f;
-    //    - chipset name (std::string)
     std::string chipsetName { "delta bridge" };
-    //    - age in years (int)
     int ageInYears = 3;
-    //3 things it can do:
-    //    - communicate with memory
+
     void communicateWithMemory(HardDrive hardDrive); 
-    //    - communicate with video card
     void communicateWithVideoCard(Monitor monitor); 
-    //    - communicate with peripherals  
     void communicateWithPeripheral(Keyboard keyboard); 
 };
 
+void Cpu::communicateWithMemory(HardDrive hardDrive)
+{
+    hardDrive.ageInYears++;
+}
+
+void Cpu::communicateWithVideoCard(Monitor monitor)
+{
+    monitor.numberOfPixels++;
+}
+
+void Cpu::communicateWithPeripheral(Keyboard keyboard)
+{
+    keyboard.numberOfKeys++;
+}
+
 struct Computer
 {
-    //5 properties:
-    //    - cd drive
     CdDrive cdDrive;
-    //    - monitor
     Monitor monitor;
-    //    - hard drive
     HardDrive hardDrive;
-    //    - keyboard
     Keyboard keyboard;
-    //    - cpu
     Cpu cpu;
-    //3 things it can do:
-    //    - load documents
+
     void loadDocument(std::string filePath); 
-    //    - save documents
     void saveDocument(std::string filePath); 
-    //    - play videos  
     void playVideo(std::string videoPath); 
 };
 
+void Computer::loadDocument(std::string filePath)
+{
+    filePath = "c:/myfiles/taxes.doc";
+}
+
+void Computer::saveDocument(std::string filePath)
+{
+    filePath = "c:/myfiles/cancel.doc";
+}
+
+void Computer::playVideo(std::string videoPath)
+{
+    videoPath = "c:/myvideos/kickflip.mpg";
+}
```

Comment:
Don't overwrite function parameters. What is the point of passing in information to the function if you're just going to discard it by overwriting it?
- [ ] revise